src/Action.php

```diff
@@ -60,6 +60,28 @@ public function __construct(BaseAction &$action)
     }
 
     /**
+     * Returns the path for an action source file.
+     *
+     * @param string $filename
+     * @return string
+     * @throws \Exception
+     */
+    private function getSourcePath(string $filename): string
+    {
+        $path = implode(DIRECTORY_SEPARATOR, [
+            dirname($_SERVER['SCRIPT_NAME']),
+            'actions',
+            "$filename.php"
+        ]);
+
+        if (!is_readable($path)) {
+            throw new \Exception('Source file not readable');
```

Comment:
I agree on using `error()`, but the exception is to interrupt the file requirement without the need to check the return. If I remove the Exception I'd have to return an empty string and change to this:

```php
if ($path = $this->getSourcePath($this->action->getActionName())) {
    $callback = require $path;
} else {
    return $this;
}
```

The catch statement would remain to catch other potential errors, so I would prefer to use it.